src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest2.cs

```diff
@@ -105,14 +105,30 @@ public void CompressedWithoutLinker ()
 		}
 
 		[Test]
-		public void BuildBasicApplication ([ValueSource (nameof (SupportedTargetFrameworks))] string tfv, [Values (true, false)] bool isRelease)
+		public void BuildBasicApplication ([ValueSource (nameof (SupportedTargetFrameworks))] string tfv, [Values (true, false)] bool isRelease, [Values ("", "en_US.UTF-8", "sv_SE.UTF-8")] string langEnvironmentVariable)
 		{
-			var proj = new XamarinAndroidApplicationProject {
-				IsRelease = isRelease,
-				TargetFrameworkVersion = tfv,
-			};
-			using (var b = CreateApkBuilder ()) {
-				Assert.IsTrue (b.Build (proj), "Build should have succeeded.");
+			string? oldLangVar = null;
+
+			if (!String.IsNullOrEmpty (langEnvironmentVariable)) {
+				// See: https://github.com/xamarin/xamarin-android/pull/7941
+				oldLangVar = Environment.GetEnvironmentVariable ("LANG");
+
+				// This will **not** call `setenv(3)`, but it should be fine since we're after the managed code here, not native
+				Environment.SetEnvironmentVariable ("LANG", langEnvironmentVariable);
+			}
+
+			try {
+				var proj = new XamarinAndroidApplicationProject {
+					IsRelease = isRelease,
+					TargetFrameworkVersion = tfv,
+				};
+				using (var b = CreateApkBuilder ()) {
+					Assert.IsTrue (b.Build (proj), "Build should have succeeded.");
```

Comment:
Can we use the `Build()` parameter here to pass environment variables in?

Example: https://github.com/xamarin/xamarin-android/blob/90afbdd2b89fc0b46ffcd81cb55ad2b85a337bc7/tests/MSBuildDeviceIntegration/Tests/InstallTests.cs#L460